sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -82,7 +82,7 @@ abstract class ExternalCatalog {
    * Note: If the underlying implementation does not support altering a certain field,
    * this becomes a no-op.
    */
-  def alterTable(db: String, tableDefinition: CatalogTable): Unit
+  def alterTable(tableDefinition: CatalogTable): Unit
```

Comment:
Logically a table is referenced by a pair of `dbName` and `tableName`. If we have a table `db1.tbl1`, and then alter table `db2.tbl1`, the semantic is not moving `tbl1` from `db1` to `db2`, but alter a nonexistent table `db2.tabl1`. For ALTER TABLE, I think there is no such a concept about `moving database`. what do you think?